app/models/spree/sale_product.rb

```diff
@@ -9,6 +9,11 @@ class SaleProduct < ActiveRecord::Base
     validates :active_sale_event_id, :product_id, :presence => true
     validates :active_sale_event_id, :uniqueness => { :scope => :product_id, :message => I18n.t('spree.active_sale.event.sale_product.already_exists') }
 
+    def self.search_product(name)
+      joins("INNER JOIN spree_product_translations ON spree_sale_products.product_id=spree_product_translations.spree_product_id")
+        .where("spree_product_translations.name LIKE '%#{name}%'")
```

Comment:
@sokmesakhiev use this to avoid SQL injection.
 .where([ "spree_product_translations.name LIKE ?", "%#{name}%" ])